include/oneapi/tbb/detail/_template_helpers.h

```diff
@@ -184,6 +184,15 @@ class raii_guard {
 public:
     raii_guard( Func f ) : my_func(f), is_active(true) {}
 
+    raii_guard (raii_guard&& g) : my_func(g.my_func), is_active(g.is_active) {
+        g.is_active = false;
```

Comment:
BTW what if copy/move of the function throws an exception ?

Reply:
In such a case, the logic would be broken and the `on_exception()` call may be performed without any exception in `try_call()` inside the temporary object destructor. But I don't know how to handle it, maybe we can use `std::move_if_noexcept`, but it still doesn't fix the case when the copy construction of the functor throws an exception.

Also, I think that this is a corner case, since it is an internal interface, and most of the use cases just use lambdas with reference capture or simple callables which do not produce exceptions.

Reply:
It was decided to check it using `static_assert`